### ISSUE TITLE
StringIndexOutOfBoundsException thrown on incomplete Accept-Language header

### ISSUE THREAD

**Issue by jebeaudet** [eclipse/jetty.project#1351]
Tested on 9.3.16.v20170120.

If you send a request with an incomplete `Accept-Language` header, Jetty will throw a `StringIndexOutOfBoundsException` on the `HttpServletRequest::getLocale()` method. 

Easy to reproduce, send any request with the header `Accept-Language: en-US,en;q=0.8,en-GB;`. The key here is the trailing `;` where the code is expecting a `q=something` I'm guessing. I've attached the relevant part of the stack trace.

I'd say the expected behavior should be to assume that since no `q` is present, take a `q=1` and just ignore the trailing `;` or at least throw a typed exception that can be translated to the user as a 400 bad request (I'm more in favor of the former solution though).

Thanks!

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.StringBuffer.charAt(StringBuffer.java:202)
	at org.eclipse.jetty.http.QuotedQualityCSV.parsedParam(QuotedQualityCSV.java:70)
	at org.eclipse.jetty.http.QuotedCSV.addValue(QuotedCSV.java:156)
	at org.eclipse.jetty.http.QuotedQualityCSV.addValue(QuotedQualityCSV.java:54)
	at org.eclipse.jetty.http.HttpFields.getQualityCSV(HttpFields.java:441)
	at org.eclipse.jetty.server.Request.getLocale(Request.java:858)
```

**Comment by gregw** [comment]
I fixed by treating missing quality as q=1.0

**Comment by karussell** [comment]
If this was fixed in 9.3.x it should be also fixed in 9.4.2.v20170220? But it occur to us - see #1697, so I wonder if this is really fixed when the parameter 'buffer' is empty?
